doc/benchmarking.rst

```diff
@@ -0,0 +1,160 @@
+Benchmarking
+============
+Performing a benchmark of fgpu or xbgpu under realistic conditions (in
+particular, with data transmitted over UDP) is difficult because there is no
+flow control, and the achievable rate can only be observed indirectly by
+running the system at a particular rate and checking whether it keeps up. We
+wish to know the largest rate at which we keep up most of the time; we'll call
+this the "critical" rate. The :file:`scratch/benchmarks` directory contains a
+script to help estimate the critical rate for fgpu (xbgpu support may be added
+later).
+
+To run it, you'll need
+
+- two servers (one to generate data and one to receive it)
+
+  - configured for use with :external+spead2:doc:`spead2's ibverbs support
+    <py-ibverbs>` on a high-speed network between them;
+  - with a Docker daemon;
+
+- somewhere to run the script (referred to below as the "client"), set up for
+  key-based SSH access to the servers in accounts that can invoke the Docker
+  client.
+- a Docker registry containing the katgpucbf Docker image.
+
+The assignment of threads to processor cores on the servers has been
+optimised for AMD Epyc servers with at least 16 cores; for other servers or
+for lower core counts, the benchmark script may need to be tuned to provide
+sensible placement.
+
+The client machine will also need a few Python packages installed, including
+:mod:`aiohttp`, :mod:`asyncssh`, :mod:`numpy` and :mod:`prometheus_client`
+(it's easiest to just run the script then install whatever packages are
+missing).
```

Comment:
Is this in lieu of a currently-absent requirements file?

Reply:
Yes.

Reply:
I wouldn't be opposed to adding another `extra` to `setup.cfg` but I don't think it's necessary to roll them into `requirements-dev.txt` since as you point out this will seldom be used.

But leaving it as you've got it here is probably also fine.

Reply:
I think I'm going to add a separate requirements.in/txt in scratch/benchmarks, similar to qualification/requirements.txt. I won't add it to setup.cfg because these benchmarks are standalone from the katgpucbf package (unlike qualification, which imports some bits of it).